boards/arm/stm32vl_disco/pinmux.c

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2016 Open-RnD Sp. z o.o.
+ * Copyright (c) 2020 Jonas Eriksson, Up to Code AB
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <device.h>
+#include <init.h>
+#include <drivers/pinmux.h>
+#include <sys/sys_io.h>
+
+#include <pinmux/stm32/pinmux_stm32.h>
+
+/* pin assignments for STM32 Discovery VL board */
+static const struct pin_config pinconf[] = {
+#ifdef CONFIG_UART_1
+	{ STM32_PIN_PA9,  STM32F1_PINMUX_FUNC_PA9_USART1_TX },
+	{ STM32_PIN_PA10, STM32F1_PINMUX_FUNC_PA10_USART1_RX },
+#endif  /* CONFIG_UART_1 */
+#ifdef CONFIG_UART_2
+	{ STM32_PIN_PA2, STM32F1_PINMUX_FUNC_PA2_USART2_TX },
+	{ STM32_PIN_PA3, STM32F1_PINMUX_FUNC_PA3_USART2_RX },
+#endif  /* CONFIG_UART_2 */
+#ifdef CONFIG_UART_3
+	{ STM32_PIN_PB10, STM32F1_PINMUX_FUNC_PB10_USART3_TX },
+	{ STM32_PIN_PB11, STM32F1_PINMUX_FUNC_PB11_USART3_RX },
+#endif  /* CONFIG_UART_3 */
+#ifdef CONFIG_PWM_STM32_1
+	{ STM32_PIN_PA8, STM32F1_PINMUX_FUNC_PA8_PWM1_CH1 },
+#endif /* CONFIG_PWM_STM32_1 */
+#ifdef CONFIG_SPI_1
+#ifdef CONFIG_SPI_STM32_USE_HW_SS
+	{ STM32_PIN_PA4, STM32F1_PINMUX_FUNC_PA4_SPI1_MASTER_NSS },
+#endif /* CONFIG_SPI_STM32_USE_HW_SS */
+	{ STM32_PIN_PA5, STM32F1_PINMUX_FUNC_PA5_SPI1_MASTER_SCK },
+	{ STM32_PIN_PA6, STM32F1_PINMUX_FUNC_PA6_SPI1_MASTER_MISO },
+	{ STM32_PIN_PA7, STM32F1_PINMUX_FUNC_PA7_SPI1_MASTER_MOSI },
+#endif /* CONFIG_SPI_1 */
+#ifdef CONFIG_SPI_2
+#ifdef CONFIG_SPI_STM32_USE_HW_SS
+	{ STM32_PIN_PB12, STM32F1_PINMUX_FUNC_PB12_SPI2_MASTER_NSS },
+#endif /* CONFIG_SPI_STM32_USE_HW_SS */
+	{ STM32_PIN_PB13, STM32F1_PINMUX_FUNC_PB13_SPI2_MASTER_SCK },
+	{ STM32_PIN_PB14, STM32F1_PINMUX_FUNC_PB14_SPI2_MASTER_MISO },
+	{ STM32_PIN_PB15, STM32F1_PINMUX_FUNC_PB15_SPI2_MASTER_MOSI },
+#endif /* CONFIG_SPI_2 */
+#ifdef CONFIG_I2C_1
+	{ STM32_PIN_PB6, STM32F1_PINMUX_FUNC_PB6_I2C1_SCL },
+	{ STM32_PIN_PB7, STM32F1_PINMUX_FUNC_PB7_I2C1_SDA },
+#endif /* CONFIG_I2C_1 */
+#ifdef CONFIG_I2C_2
+	{ STM32_PIN_PB10, STM32F1_PINMUX_FUNC_PB10_I2C2_SCL },
+	{ STM32_PIN_PB11, STM32F1_PINMUX_FUNC_PB11_I2C2_SDA },
+#endif /* CONFIG_I2C_2 */
+#ifdef CONFIG_ADC_1
```

Comment:
adc not enabled in board dts.

Reply:
Hmm, I should get it from `dts/arm/st/f1/stm32f1.dtsi` if I am not mistaken?

Reply:
Except GPIOs, all peripheral devices are disabled in .`dtsi` files.
So if you want to enable on the board, you should enable the node in board .dts file.
Also, you should make the appropriate setting in board kconfig.defconfig

Reply:
Yes, I was under the impression that you
1. Enable it in the board DTS
2. Set CONFIG_ADC_1 in the board Kconfig
I essentially copied the behavior from nucleo_f103rb seeing as that support was probably more "modern" than the disco boards, but poking around the *_disco directories i see that those leave the ADC out like you say. I've removed it now.

Reply:
ok, seems that nucleo_f103b should actually be fixed.

Reply:
That explains my confusion at least, thanks :-)